src/github.com/getlantern/flashlight/config/config.go

```diff
@@ -375,13 +287,6 @@ func (cfg *Config) ApplyDefaults() {
 	if cfg.TrustedCAs == nil || len(cfg.TrustedCAs) == 0 {
 		cfg.TrustedCAs = defaultTrustedCAs
 	}
-
-	if cfg.Client.DeviceID == "" {
-		// There is no true privacy or security in instance ID.  For that, we rely on
-		// transport security.  Hashing MAC would buy us nothing, since the space of
-		// MACs is trivially mapped, especially since the salt would be known
-		cfg.Client.DeviceID = base64.StdEncoding.EncodeToString(uuid.NodeID())
-	}
```

Comment:
Moved this. Previously, if cfg.Client was nil, this would create a panic. Moved to within applyClientDefaults along with everything else that changes the client defaults.